datasets/swda/swda.py

```diff
@@ -169,35 +423,356 @@ def _split_generators(self, dl_manager):
+            for i_utt, utt in enumerate(trans.utterances):
+                id_ = i_trans + i_utt
+                yield id_, {
+                    "swda_filename": utt.swda_filename,
+                    "ptb_basename": utt.ptb_basename,
+                    "conversation_no": utt.conversation_no,
+                    "transcript_index": utt.transcript_index,
+                    "act_tag": utt.act_tag,  # Need change to labels
+                    "damsl_act_tag": utt.damsl_act_tag(),
+                    "caller": utt.caller,
+                    "utterance_index": utt.utterance_index,
+                    "text": utt.text,
+                    "pos": utt.pos,
+                    "trees": utt.trees,
+                    "ptb_treenumbers": utt.ptb_treenumbers,
+                }
 
-                # Parse row into speaker info | utterance text | dialogue act tag.
-                parsed_row = row.rstrip("\r\n").split("|")
 
-                # Also returning dialogue act integer label.
-                yield f"{split}-{id_}", {
-                    "speaker": parsed_row[0],
-                    "utterance_text": parsed_row[1],
-                    "dialogue_act_tag": parsed_row[2],
-                }
+class CorpusReader:
+    """Class for reading in the corpus and iterating through its values."""
+
+    def __init__(self, src_dirname, split_file=None):
+        """
+        Reads in the data from `src_dirname` (should be the root of the
+        corpus).  Assumes that the metadata file `swda-metadata.csv` is
+        in the main directory of the corpus, using that file to build
+        the `Metadata` object used throughout.
+
+        Added split_files to separate between splits. If given None it will use all data files.
+        """
+        self.src_dirname = src_dirname
+        metadata_filename = os.path.join(src_dirname, "swda-metadata.csv")
+        self.metadata = Metadata(metadata_filename)
+        self.split_file = split_file
+
+    def iter_transcripts(self, display_progress=True):
+        """
+        Iterate through the transcripts.
+
+        Parameters
+        ----------
+        display_progress : bool (default: True)
+            Display an overwriting progress bar if True.
+        """
+
+        # All files names.
+        filenames = glob.glob(os.path.join(self.src_dirname, "sw*", "*.csv"))
+        # If no split files are mentioned just use all files.
+        self.split_file = filenames if self.split_file is None else self.split_file
+        # Filter out desired file names
+        filenames = [
+            file for file in filenames if os.path.basename(file).split("_")[-1].split(".")[0] in self.split_file
+        ]
+        i = 1
+        for filename in filenames:
+            # Optional progress bar:
+            if display_progress:
+                sys.stderr.write("\r")
+                sys.stderr.write("transcript %s" % i)
+                sys.stderr.flush()
```

Comment:
Can you remove the progress bar ? (although it's not used if I understand correctly ?)

There's already one added by the `datasets` library when examples are generated.
(also we don't use `sys` for printing such things)